### PR TITLE
Updated the javadocs

### PR DESCRIPTION
**Pull request by szczepiq**
Polishing the javadocs, starting with this small PR:

Issue #596
- updated @since to 2.1.0 (instead 2.0.0)
- added more docs around the verification collector

### CODE REVIEW

**Comment by TimvdLippe** [comment]
LGTM

File: src/main/java/org/mockito/internal/stubbing/defaultanswers/ReturnsMoreEmptyValues.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 import java.lang.reflect.Array;
 
 /**
- * It's likely this implementation will be used by default by every Mockito 2.0.0 mock.
+ * It's likely this implementation will be used by default by every Mockito 2.1.0 mock.
```
Comment (bric3) [inline src/main/java/org/mockito/internal/stubbing/defaultanswers/ReturnsMoreEmptyValues.java]: In the end this is not the case.

Comment (szczepiq) [inline src/main/java/org/mockito/internal/stubbing/defaultanswers/ReturnsMoreEmptyValues.java]: Good catch. 3.0.0? :)

Comment (bric3) [inline src/main/java/org/mockito/internal/stubbing/defaultanswers/ReturnsMoreEmptyValues.java]: Why not, you advocated on the google code tracker the change. I forgot about it too. I think this answer is good.

If you agree and I agree too then let's make the change ;)

Comment (szczepiq) [inline src/main/java/org/mockito/internal/stubbing/defaultanswers/ReturnsMoreEmptyValues.java]: I agree. It's a good change I think. However, it's a scope creep and needs documentation update. Also many tests may fail :) Do you want to include it in 2.1.0?

Comment (bric3) [inline src/main/java/org/mockito/internal/stubbing/defaultanswers/ReturnsMoreEmptyValues.java]: I meant for 3.0.0 ;)
